src/lib/eslint/ESLintEngine.ts

```diff
@@ -21,74 +109,137 @@ export class ESLintEngine implements RuleEngine {
+		// change to treat targetPaths which are directories as project directories, and look inside them for a
+		// tsconfig.json file.  Until then, just require that if you want to run against *.ts, you better do so from a
+		// working directory that contains tsconfig.
+		this.esconfig = await this.fileHandler.exists("tsconfig.json") ?
+			DEFAULT_ESCONFIG_TS : DEFAULT_ESCONFIG;
+
 		this.initialized = true;
 	}
 
 	getCatalog(): Promise<Catalog> {
-		return Promise.resolve({rulesets: [], categories: [], rules: []});
+		const categoryMap: Map<string, RuleGroup> = new Map();
+		const catalog: Catalog = {rulesets: [], categories: [], rules: []};
+		const rules: Rule[] = [];
+		const cli = new CLIEngine(this.esconfig);
+		cli.getRules().forEach((rule: ESRule, key: string) => {
+			const docs = rule.meta.docs;
+			const categoryName = docs.category;
+			let category = categoryMap.get(categoryName);
+			if (!category) {
+				category = {name: categoryName, engine: ESLintEngine.NAME, paths: []};
+				categoryMap.set(categoryName, category);
+			}
+			category.paths.push(docs.url);
+
+			const r = {
+				engine: ESLintEngine.NAME,
+				sourcepackage: ESLintEngine.NAME,
+				name: key,
+				description: docs.description,
+				categories: [docs.category],
+				rulesets: [docs.category],
+				languages: [ESLintEngine.JAVASCRIPT_LANGUAGE],
+				defaultEnabled: docs.recommended,
+				url: docs.url
+			};
+			if (key.startsWith("@typescript")) {
+				// Typescript is superset of javascript, so add it on top here.
+				r.languages.push(ESLintEngine.TYPESCRIPT_LANGUAGE);
+			}
+			rules.push(r)
+		});
+
+		catalog.categories = Array.from(categoryMap.values());
+		catalog.rules = rules;
+		return Promise.resolve(catalog);
 	}
 
 	public async getAll(): Promise<Rule[]> {
-		// PmdCatalogWrapper is a layer of abstraction between the commands and PMD, facilitating code reuse and other goodness.
-		this.logger.trace('Getting PMD rules.');
+		this.logger.trace('Getting eslint rules.');
 		return Promise.resolve([]);
 	}
 
-	public async run(paths: PathGroup[], target: string[]): Promise<string> {
-		this.logger.trace(`About to run eslint rules. Target count: ${target.length}, filter count: ${paths.length}`);
-		try {
-			/*
-			const linter = new Linter();
-
-			const sample =
-				{
-					fatal: false,
-					ruleId: "semi",
-					severity: 2,
-					line: 1,
-					column: 23,
-					message: "Expected a semicolon.",
-					fix: {
-						range: [1, 15],
-						text: ";"
-					}
-				};
-
-			const messages = linter.verify("var foo;", {
-				rules: {
-					semi: 2
-				}
-			}, {filename: "foo.js"});
-*/
+	public async run(ruleGroups: RuleGroup[], rules: Rule[], targets: string[]): Promise<RuleResult[]> {
+		const targetPaths: string[] = targets;
+		// If we didn't find any paths, we're done.
+		if (targetPaths == null || targetPaths.length === 0) {
+			this.logger.trace('No matching target files found. Nothing to execute.');
```

Comment:
I feel like the message should indicate that the engine is ESLint.

Reply:
Also, I feel we should be surfacing this information to users.

Reply:
I added the word 'eslint' to the message about no target files.

Roopa: this is trace info, not an error condition.  It will happen all the time, say when someone just runs the scanner against a .cls file.  The engines do not always match the target files given, so it could be the engine does nothing.  The trace message is just there for debugging in the field, like the others are.